lib/facility.rb

```diff
@@ -1,14 +1,45 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(hash)
+    # (hash) or (phonebook)
+    @name = hash[:name]
+    @address = hash[:address]
+    @phone = hash[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration") 
+      @registered_vehicles << vehicle
+    end
+    collect_fees(vehicle)
```

Comment:
Nice job using a helper method here to do this logic for you! I would have put this inside the if block though because we wouldn't want to collect any fees if we can't actually register the vehicle.